extensions/amp-recaptcha-input/0.1/amp-recaptcha-input.js

```diff
@@ -13,3 +13,74 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+/**
+ * @fileoverview Async Input Element that uses the
+ * amp-recaptcha-service to dispatch actions, and return
+ * recaptcha tokens
+ */
+
+import {CSS} from '../../../build/amp-recaptcha-input-0.1.css';
+import {Layout} from '../../../src/layout';
+import {
+  installRecaptchaService,
+  recaptchaServiceFor,
+} from './amp-recaptcha-service';
+import {isExperimentOn} from '../../../src/experiments';
+import {setStyles, toggle} from '../../../src/style';
+
+/** @const */
+const TAG = 'amp-recaptcha-input';
+
+export class AmpRecaptchaInput extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!./amp-recaptcha-service.AmpRecaptchaService} */
+    this.recaptchaService_ = recaptchaServiceFor(this.win);
+
+    /** @private {boolean} */
+    this.isExperimentEnabled_ = isExperimentOn(this.win, 'amp-recaptcha-input');
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /** @override */
+  buildCallback() {
+    if (!this.isExperimentEnabled_) {
+      return;
+    }
+
+    return this.mutateElement(() => {
+      toggle(this.element);
+      setStyles(this.element, {
+        'position': 'absolute',
+        'width': '1px',
+        'height': '1px',
+        'overflow': 'hidden',
+        'visibility': 'hidden',
+      });
+    });
+  }
+
+  /** @override */
+  layoutCallback() {
+    return this.recaptchaService_.register(this.element);
```

Comment:
There is a race condition here. 
It's possible that runtime scheduled layout for the element first. And then elements owner scheduled layout for it again. We should not register the service twice in this case. Please refer the `<amp-analytics>` `layoutCallback()`.
Let's put a TODO here and fix the race condition in a following PR.